lib/Doctrine/Migrations/MigrationPlanCalculator.php

```diff
@@ -15,17 +16,21 @@
 use Doctrine\Migrations\Metadata\Storage\MetadataStorage;
 use Doctrine\Migrations\Version\Direction;
 use Doctrine\Migrations\Version\Version;
+use function array_diff;
 use function array_filter;
 use function array_map;
 use function array_reverse;
+use function count;
+use function in_array;
+use function reset;
 
 /**
  * The MigrationPlanCalculator is responsible for calculating the plan for migrating from the current
  * version to another version.
  *
  * @internal
  */
-final class MigrationPlanCalculator
+class MigrationPlanCalculator
```

Comment:
Not happy with opening the method for extension, but I trust that this is the most sensible way to do this. I'm not sure if extracting an interface for the purpose of mocking would be sensible.

Reply:
`MigrationPlanCalculator` will be refactored in a upcoming PR. Currently `getNewMigrations` and `getExecutedUnavailableMigrations` dont belong conceptually to this class. After extracting those methods, I will introduce an interface and mark the class as final again.

Reply:
this has been fixed in https://github.com/doctrine/migrations/pull/897, now the mock is done on the interface and the class is back to final